### PR TITLE
Add "What Makes a Good Vector Database" article

### PR DESCRIPTION
**Pull request by SJZhung**

### CODE REVIEW

**Comment by live9080** [comment]
Can you please rebase or merge the latest 'dev' branch? We have added an action in the 'dev' branch that automatically generates a preview URL. After you update your PR, you should be able to get the preview URL added by the bot(like the screenshot below) so that reviewers can access the preview.

image

**Comment by live9080** [comment]
Could you please rebase or merge the latest dev branch? We've added an action in the dev branch to automatically generate a preview URL. After you update the PR, you'll get a preview URL added by the bot (just like the screenshot below) so that reviewers can take a look.

image

File: articles/What Makes a Good Vector Database/README.md [hunk 1]
```diff
@@ -0,0 +1,442 @@
+---
+title: "What Makes a Good Vector Database --- LanceDB vs Pinecone"
+author: ["Alex Hu / Machine Learning Engineer", "Steven Zhang / Backend Software Engineer", "Anjing Wang / Full Stack Engineer"]
+createTime: 2024-05-22
+tags: ["Vector Database", "LanceDB", "Pinecone"]
+thumb: "./thumb.png"
+thumb_h: "./thumb_h.png"
```
Comment (live9080) [inline articles/What Makes a Good Vector Database/README.md:7]: The `thumb_h` is not existing. Could you upload it please? Thanks

File: articles/What Makes a Good Vector Database/README.md [hunk 2]
```diff
@@ -0,0 +1,442 @@
...
+table = db.create_table("pets", schema=Pets)
+result = table.search("dog")
+p = Path("path/to/images/samoyed_100.jpg")
+query_image = Image.open(p)
+table.search(query_image)
+```
+
+### 3.3. Search
+
+Pinecone Database it empowers users to perform scalable Approximate
```
Comment (wholehope) [inline articles/What Makes a Good Vector Database/README.md:191]: remove it

File: articles/What Makes a Good Vector Database/README.md [hunk 3]
```diff
@@ -0,0 +1,442 @@
...
+Nearest Neighbor (ANN) searches quickly and accurately, excels with both
+sparse and dense vector support, allowing for mixed searches in a
+singular index. This hybrid approach combines semantic depth with
+keyword precision for improved relevance and accuracy in results.
+Additionally, Pinecone enables the attachment of metadata key-value
+pairs to indexed vectors, coupled with detailed filter expressions
+during queries, enabling tailored searches based on specific criteria.
+With support for up to 40 kb of metadata per vector, [[selective
+metadata
+indexing]{.underline}](https://docs.pinecone.io/docs/manage-indexes/#selective-metadata-indexing)
```
Comment (wholehope) [inline articles/What Makes a Good Vector Database/README.md:201]: format error

File: articles/What Makes a Good Vector Database/README.md [hunk 4]
```diff
@@ -0,0 +1,442 @@
...
+is advised to prevent unnecessary high-cardinality metadata from being
+indexed when not required for filtering. This adaptability allows users
+to create diverse search experiences, from simple text searches to
+complex queries.
+
+Pinecone sparse-dense vectors have the following limitations:
+- Pinecone supports sparse vector values of sizes up to 1000 non-zero
+ values.
+- Pinecone only supports upserting sparse-dense vectors to p1 and s1 indexes.
+- In order to query an index using sparse values, the index must use the [[dot product> metric]{.underline}](https://docs.pinecone.io/docs/indexes#distance-metrics). Attempting to query any other index with sparse values returns an error.
```
Comment (wholehope) [inline articles/What Makes a Good Vector Database/README.md:211]: same format error

**Pull request merged**